src/MoBi.Core/Domain/Repository/BuildingBlockRepository.cs

```diff
@@ -1,43 +1,83 @@
 using System.Collections.Generic;
 using System.Linq;
-using OSPSuite.Utility.Collections;
 using MoBi.Core.Domain.Model;
 using MoBi.Core.Services;
+using OSPSuite.Core.Domain;
 using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Utility.Collections;
 
 namespace MoBi.Core.Domain.Repository
 {
    public interface IBuildingBlockRepository : IRepository<IBuildingBlock>
    {
-      IEnumerable<TBuildingBlock> All<TBuildingBlock>() where TBuildingBlock : IBuildingBlock;
+      IReadOnlyList<MoleculeBuildingBlock> MoleculeBlockCollection { get; }
+      IReadOnlyList<MoBiReactionBuildingBlock> ReactionBlockCollection { get; }
+      IReadOnlyList<ExpressionProfileBuildingBlock> ExpressionProfileCollection { get; }
+      IReadOnlyList<IndividualBuildingBlock> IndividualsCollection { get; }
+      IReadOnlyList<PassiveTransportBuildingBlock> PassiveTransportCollection { get; }
+      IReadOnlyList<MoBiSpatialStructure> SpatialStructureCollection { get; }
+      IReadOnlyList<ObserverBuildingBlock> ObserverBlockCollection { get; }
+      IReadOnlyList<EventGroupBuildingBlock> EventBlockCollection { get; }
+      IReadOnlyList<InitialConditionsBuildingBlock> InitialConditionBlockCollection { get; }
+      IReadOnlyList<ParameterValuesBuildingBlock> ParametersValueBlockCollection { get; }
+      IndividualBuildingBlock IndividualByName(string buildingBlockName);
+      ExpressionProfileBuildingBlock ExpressionProfileByName(string buildingBlockName);
    }
 
    public class BuildingBlockRepository : IBuildingBlockRepository
```

Comment:
The new functionality of finding a project's building blocks is concentrated here. Most tasks formerly used IMoBiContext to get the project then just used ```AllBuildingBlocks```